### PR TITLE
Add submodule to set up ACM

### LINKED ISSUE #268: Add submodule to set up ACM
We should add a new submodule which can be used to [install ACM](https://cloud.google.com/anthos-config-management/docs/how-to/installing) on a cluster.

This would involve:
1. Deploying the ACM operator
2. Automatically creating an [SSH keypair](https://www.terraform.io/docs/providers/tls/r/private_key.html) for the operator, uploading it to the cluster (as a secret)
3. Configuring the operator by creating a `ConfigManagement` resource in the cluster with these required values filled in:
- spec.clusterName
- spec.git.syncRepo
- spec.git.secrectType (hardcode to ssh to start)
- spec.git.policyDir

The module inputs should include:
- cluster ID/self-link
- sync repo
- policy directory
- sync branch (default master)

Outputs should include:
- SSH public key (to give the cluster access to git)

### PR DESCRIPTION
**Pull request by kopachevsky**
Initial commit

Fixes #268

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -50,3 +50,4 @@ credentials.json
 
 # File to populate env vars used by Docker test runs
 .envrc
+/modules/acm/config-management-operator.yaml
```
Comment (paulpalamarchuk) [inline .gitignore]: It has to be absolute path?

Comment (kopachevsky) [reply .gitignore]: no, wil lfix

**Comment by kopachevsky** [comment]
> Add description to of using ACM

I think usage docs is out of the scope, added readme with installation steps

**Comment by kopachevsky** [comment]
@morgante @aaron-lane I do have issue with destroy here, it seems simple_zonal cluster begin to destroy before/same time as acm module, and commands for acm destroy like `kubectl delete` just can't reach cluster, do you know any hints about how to force module to be destroyed first?

**Comment by kopachevsky** [comment]
> Is there a simple test we can add to to the simple-zonal suite to verify the validity of the ACM submodule?

@aaron-lane I was thinking about verification test but only issue is ACM will be fully workable after user will take ssh_key from module outputs and add it to his profile, possible options here:

1. verify that pods and operators are created or not
2. refactore ACM module to support config git repos access with no authentication (https:// protocol)

File: modules/acm/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+In order to use this module you must have Service Account with roles listed [Terraform Kubernetes Engine Module](../../README.md)
+plus **roles/container.admin** role.
+
+## Usage example
+
+See [examples/simple_zonal](../../examples/simple_zonal) cluster example.
+
+## Installation
+
+Module automate installation steps described at [Installing Anthos Config Management](https://cloud.google.com/anthos-config-management/docs/how-to/installing) page.
```
[the remark this answers is not in the document]
Comment (paulpalamarchuk) [reply modules/acm/README.md]: Updated

File: modules/acm/README.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+Git access provisioned via "Using an SSH keypair" option, to finish authorization add SSH public from **git\_creds\_public**
```
[the remark this answers is not in the document]
Comment (paulpalamarchuk) [reply modules/acm/README.md]: Changed

File: modules/acm/README.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+output variable to your account at your Git server.
```
[the remark this answers is not in the document]
Comment (paulpalamarchuk) [reply modules/acm/README.md]: Removed

File: test/task_helper_functions.sh [hunk 5]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env bash
+#!/usr/bin/env bash{
```
[the remark this answers is not in the document]
Comment (paulpalamarchuk) [reply test/task_helper_functions.sh]: Fixed

**Comment by paulpalamarchuk** [comment]
PR rebased. Fixed issue with gsutil credentials.

File: build/int.cloudbuild.yaml [hunk 6]
```diff
@@ -130,7 +130,7 @@ steps:
   waitFor:
     - create simple-zonal-local
   name: 'gcr.io/cloud-foundation-cicd/$_DOCKER_IMAGE_DEVELOPER_TOOLS:$_DOCKER_TAG_VERSION_DEVELOPER_TOOLS'
-  args: ['/bin/bash', '-c', 'source /usr/local/bin/task_helper_functions.sh && kitchen_do converge simple-zonal-local']
+  args: ['/bin/bash', '-c', 'source /usr/local/bin/task_helper_functions.sh && env > /dev/null && kitchen_do converge simple-zonal-local']
```
Comment (aaron-lane) [inline build/int.cloudbuild.yaml]:
```suggestion
  args: ['/bin/bash', '-c', 'source /usr/local/bin/task_helper_functions.sh && kitchen_do converge simple-zonal-local']
```

File: modules/acm/main.tf [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  cluster_endpoint       = "https://${var.cluster_endpoint}"
```
Comment (thecodejunkie) [inline modules/acm/main.tf]: Couldn't this be fetched from `data.google_container_cluster.primary.endpoint` as per the documentation https://www.terraform.io/docs/providers/google/d/google_container_cluster.html ?

Comment (aaron-lane) [reply modules/acm/main.tf]: Depending on a data source for the cluster endpoint breaks the dependency graph if this module is in the same configuration as the cluster. We witnessed the cluster being destroyed before the resources dependent on the endpoint.

**Pull request merged**